server/controllers/ArticleController.js

```diff
@@ -24,26 +24,40 @@ class ArticleController {
    */
   static createArticle(req, res, next) {
     const {
-      title, description, body,
+      title, description, body, tags,
     } = req.body.article;
 
     // generating a slug
     const id = uuid();
     const slug = slugCreate(`${title.toLowerCase()}-${id}`);
     const authorId = req.body.decoded.id;
 
-
     const {
       username, bio, image,
     } = req.user;
 
-    Article.create({
+    const articleObj = {
       id, slug, title, description, body, authorId,
-    })
-      .then(article => res.status(201).json({
-        article: articleControllerHelper
-          .createArticleResponse(article, username, bio, image, slug),
-      })).catch(next);
+    };
+
+    Promise.all([
+      Article.create(articleObj),
+      articleControllerHelper.createAndFindTags(tags),
```

Comment:
These methods `Article.create(articleObj)` and `articleControllerHelper.createAndFindTags(tags)` run asynchronously, as I understand. There is the possibility that creates article operation fails while the create tags operation already succeeds. This implies that there can be situations when an author could create tags without having his article created. Is this what you want?